### ISSUE TITLE
[FEATURE] Alpinejs x-data field is condensed to a single line

### ISSUE THREAD

**Issue by pinkfrog9** [Riverside-Healthcare/djLint#723]
For example, the following code, 
<img width="724" alt="image" src="https://github.com/Riverside-Healthcare/djLint/assets/107285219/cc23d3b2-d6e4-4991-be0b-c426a1786d27">

After formatting becomes this:
<img width="1148" alt="image" src="https://github.com/Riverside-Healthcare/djLint/assets/107285219/4597182c-ab8f-4c45-a233-1bc85b318f63">

Is there some mechanism to format the js inside x-data field. If not, is it possible to disable the format on x-data?

**Comment by christopherpickering** [comment]
Thanks! Is it always "real" js or is there sometimes template syntax?
I'm using the python version of js beauty here and it can only handle real js.
We can probably add flags for both though.

**Comment by pinkfrog9** [comment]
> Thanks! Is it always "real" js or is there sometimes template syntax?
In my use case, I use jinja2 + alpine.js.  So the x-data can contain jinja template.

**Comment by scratchmex** [comment]
is there a way do disable the formatting inside the ticks for html attributes?

**Comment by CleitonDeLima** [comment]
> Thanks! Is it always "real" js or is there sometimes template syntax?

Pure js.

**Comment by CleitonDeLima** [comment]
Any alternative to this? Could I disable the format of a specific attribute, in this case x-data?

**Comment by oliverhaas** [comment]
I think your best option is to implement something like this yourself and submit a PR.
I've looked at how to go about this, since I'd also like to have that feature for hyperscript, but it's unfortunately more than a couple of hours of work, since I think you will need to touch a couple more regexes to get to clean attribute parsing.

**Comment by CleitonDeLima** [comment]
> I think your best option is to implement something like this yourself and submit a PR. I've looked at how to go about this, since I'd also like to have that feature for hyperscript, but it's unfortunately more than a couple of hours of work, since I think you will need to touch a couple more regexes to get to clean attribute parsing.

Is it valid to think about a configuration to disable the formatting of certain attributes? I can work on it, I believe it would be less work.

**Comment by oliverhaas** [comment]
> Is it valid to think about a configuration to disable the formatting of certain attributes? I can work on it, I believe it would be less work.

Yes, that sounds great! I think a regex to match attributes the user does not want to be formatted would be perfect, so you could just do `^x-.*$` or something for the alpine stuff.

I only wanted to give a heads up that I don't think you have some point in the parsing in djlint where you have the attributes of each tag as a list of keys and values or something, so it's a bit more difficult than that, but certainly doable, and useful for other possible future features.
